### DISCUSSION TITLE
Some problems with Cluster.py

### DISCUSSION THREAD

**Discussion by zhaowb12138** [SSCHAcode/python-sscha#110] [Q&A] [answered]
Hi developers,
I have some problems with Cluster.py. Recently I updated the SSCHA,and I meet an error in Cluster.py
![image](https://user-images.githubusercontent.com/108879161/203065086-612a8896-45de-4a44-8a7c-8fe999bc6f35.png)
However, when I replaced cluster.py with the original version, this problem was removed.Have you ever meet this problem?Is this a bug or there's something wrong with the way I'm using it ?

Here is the version with no problem
[Cluster.zip](https://github.com/SSCHAcode/python-sscha/files/10056868/Cluster.zip)

**Comment by mesonepigreco** [comment, marked as answer]
Dear @zhaowb12138 ,
Thanks for reporting. Yes, the latest version of Cluster.py is compatible with a custom calculator for espresso defined in
cellconstructor.calculators.Espresso rather than the ASE one. This is because of some new features introduced in the latest version of the official ASE calculators (specifically regarding the I/O strategy with parallelization) which conflicted with our modules, generating bugs at execution time. 
For this reason, we decided to support our own calculators for espresso. It works very similarly to the ASE one. Most things should be exactly the same, and you should be able to obtain the same results just by replacing the import of the ASE espresso calculator with the ASE one. The only difference is that the input_data flag needs to state the subdictionaries 'control' and 'system' as in the espresso input file.

For a guide and an example of how to configure the new calculator, see section 3.6 of the guide: [https://raw.githubusercontent.com/SSCHAcode/python-sscha/master/python-sscha.pdf](https://raw.githubusercontent.com/SSCHAcode/python-sscha/master/python-sscha.pdf)

However, we should probably still support ASE in the cluster module as we usually do not use the broken I/O module there. I will raise an issue on GitHub to fix the compatibility in the next release. 

I suggest you anyway to update cluster.py to the new version since we fixed some bugs in the cluster module and improved its functionality, reducing the connections spawn to the clusters (speeding up the jobs submission), and please let us know if with the new cellconstructor calculator for expresso works correctly or you have problems.

Thanks again for the report,
Bests,
Lorenzo

**Reply by zhaowb12138** [reply]
Thank you for your reply. 

I configure the new calculator, and it worked. 

PS: I think there is a spelling mistake in the guidance document. It is not "import cellconstructor. Calculators", but "import cellconstructor.calculators" right?

Thanks again
Bests,
Zhao

**Reply by diegomartinez2** [reply]
Thanks Zhao, I fixed the bug you reported in the documentation.

Best regards,
Diego.
